Update contribute page (#1436)
act-rules/act-rules.github.io: pages/contribute/rule-feedback.md

---
title: Give feedback on rules
---

> Feedback on rules is encouraged on our open [GitHub repository](https://github.com/act-rules/act-rules.github.io), so you don't even have to be a member of the Community Group to contribute in this way.

Reviewer: ```suggestion
Feedback is encouraged on our open [GitHub repository](https://github.com/act-rules/act-rules.github.io), so you don't even have to be a member of the Community Group to contribute in this way.
```

I think the first paragraph on this page should cover all types of feedback, not only on rules.

> If you "watch" the repository, you will get notifications when changes are happening.

Reviewer: It might be helpful if "watch" links to something that explains how to do that.

Reviewer 2: It might actually be possible to have a link to watch automatically.
(when new repos are created in my org, Github auto-watch me and send me a link to un-watch; if link to un-watch exists, I assume link to watch should exist)

Reviewer 3: Added link to Propose Rules page.

## Feedback for ideas and early drafts (Issues on GitHub)

> Rules and Definitions (shared terms across rules) that are GitHub Issues can be everything from just a title to a quite finished rule. The issues will be prefixed with either "Rule:" (or possibly "New Rule:") or "Algorithm".

Reviewer: Are algorithms the same as definitions? Even for someone fairly familiar with things this paragraph isn't very clear.

Reviewer: ```suggestion
Draft rules and definitions (shared terms across rules) that are currently GitHub issues can be anything from just a title to an almost finished rule. These issues will be prefixed with either "Rule:" (or possibly "New Rule:") or "Algorithm".
```

Reviewer 2: I'm also not sure what "Algorithm" refers to in our process 🤔

Author: Yikes! Yeah "algorithm" is what we used to call definitions, when rules were would prescribe the exact procedure to follow.

> The top comment on an issue should hold the updated rule or algorithm, and underneath there will be a thread of comments and a history of actions related to the issue.

Reviewer: ```suggestion
The top comment on such an issue should hold the most up to date version of the draft rule or algorithm, and underneath there will be a thread of related comments and a history of actions.
```

> Please feel free to add your own comments to issues.

Reviewer: ```suggestion
Please feel free to add your own comments to these issues.
```

> Start giving feedback on issues:

Reviewer: ```suggestion
To start giving such feedback:
```

> - See the [ideas and early drafts that might need input](https://github.com/act-rules/act-rules.github.io/issues)

Reviewer 2: Should we filter that on labels such as "rule idea" or "Should we proceed?" or some other?
Many issues are problem with existing rules and not ideas/early draft, and the feedback to give is quite different…

Reviewer: That would be good, if there is a generic filter for "ideas and early drafts", which is what this section covers. It would need to include definition drafts as well as rule ideas and drafts, and anything similar.

- Use the [Definition of "Done"](../design/definition-of-done) to guide your review
- Learn more about [commenting on issues](#) (TBD)

## Feedback and reviews for draft rules (Pull Requests on GitHub)

> Pull requests are rule drafts that are ready for peer reviews. You can sign up as a reviewer and follow the rule through iterations. A rule needs three approvals, and if a rule has the label "Reviewer wanted", we need more people to sign up for reviewing it. But even if three people are already signed up, we can always use more eyes on it.

Reviewer 4: This makes it look like that before starting a review, one has to "register" for it, when all that is required is to have a GitHub account.

Reviewer: Does someone need to be a member of the community group to review a PR, or is this still just GitHub permission levels?

Reviewer: ```suggestion
Pull requests are rule drafts that are ready for peer reviews. You can assign yourself as a reviewer and follow the rule through iterations. A rule needs three approvals, and if a rule has the label "Reviewer wanted", we need more people to review it. However, even if three people are already reviewing it, we can always use more eyes on things.
```

> You can both add single comments to a pull request and do a full review.

Reviewer 2: ```suggestion
You can either add single comments to a pull request or do a full review.
```

Reviewer: Except that you can do both. So perhaps "and/or".

**Only approve a rule** if you feel confident (to the best of your knowledge) that this rule is 100% ready to be a published ACT rule. Please also refer to the [Definition of "Done"](../design/definition-of-done).

If you for some reason are not so confident, you can always leave a comment for the things that you have an opinion about, without doing a full review.

> Start giving feedback and doing reviews for draft rules:

Reviewer: ```suggestion
To start giving feedback and doing reviews for draft rules:
```

> - See the [draft rules, algorithms and more that need more reviewers](https://github.com/act-rules/act-rules.github.io/pulls?q=is%3Aopen+is%3Apr+label%3A%22reviewer+wanted%22)

Reviewer 2: Here also, I do not know what "algorithms" means in our context.

- Use the [Definition of "Done"](../design/definition-of-done) to guide your review
- Learn more about [reviewing and commenting on pull requests](#) (TBD)

## Feedback/corrections for published rules

> We welcome any input for changes, also for already published rules. The published rules should be as precise as possible, limiting the number of potential false positives and false negatives. For this it is important to draw on as many tool and methodology implementations and expert opinions as possible.

Reviewer: ```suggestion
We also welcome any feedback and suggestions for changes to already published rules. The published rules should be as precise as possible, limiting the number of potential false positives and false negatives. For this, it is important to draw on as many tool and methodology implementations and expert opinions as possible.
```

> Feedback and corrections for existing rules can target both on the sections of the rule itself (Applicability, Expectations, Accessibility Support, etc.) and the test cases, to e.g. expand the edge case coverage.

Reviewer: ```suggestion
Feedback and corrections for existing rules can target any of the sections of the rule itself (Applicability, Expectations, Accessibility Support, etc.), as well as the test cases, for example to expand the edge case coverage.
```

You have several options for correcting existing rules:

> - **Open a discussion in an Issue:** If you want to open a discussion, you can [open a new issue on GitHub](https://github.com/act-rules/act-rules.github.io/issues) repository to discuss the issue with the existing rule. Learn more about [creating issues](#) (TBD).

Reviewer: ```suggestion
- **Open a discussion in an Issue:** If you want to open a discussion, you can [open a new issue on the GitHub repository](https://github.com/act-rules/act-rules.github.io/issues) to discuss the issue with the existing rule. Learn more about [creating issues](#) (TBD).
```

Alternatively remove the word repository.

## Feedback and reviews for draft rules (Pull Requests on GitHub)

> Pull requests are rule drafts that are ready for peer review. You can assign yourself as a reviewer and follow the rule through iterations. A rule needs three approvals, and if a rule has the label "Reviewer wanted", we need more people to review it. However, even if three people are already reviewing it, we can always use more eyes on things.

Reviewer 2: "assign yourself as a reviewer" is incorrect in Github lingo.
"assign" is used by Github for the person(s) in charge of the PR, reviewers are "requested". There has been several occurrences of people "assigning" other rather than "requesting review" from them. I'm a bit afraid this may drive non Github-savy people to click the "assign" button rather than the "request review" one. Especially, there is a Github button called "assign yourself" which is **not** what we want people to click in order to be reviewers…

<img width="290" alt="assign-review" src="https://user-images.githubusercontent.com/19969498/103877088-e201f500-50d4-11eb-94e7-7443af69ad4d.png">

OTOH, I agree that "you can request a review from yourself" sounds plain wrong in regular language… 🤔 I'm not having a good suggestion here.

Reviewer 3: changed "assign yourself..." to "add yourself...", which should now be hopefully less confusing.
